Fix some PDF tests after 296104@main

rr-codes opened the pull request:
#### ce9a4fafb44ecf70b9c84a30a544748cd90eb1c5
<pre>
Fix some PDF tests after 296104@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=294482">https://bugs.webkit.org/show_bug.cgi?id=294482</a>
<a href="https://rdar.apple.com/153344673">rdar://153344673</a>

Reviewed by Abrar Rahman Protyasha.

296104@main accidentally broke some PDF tests, but it also conveniently also accidentally
disabled those same tests because it didn&apos;t replace the PDFKIT_FOR_TESTING flag everywhere.

Fix that, and then fix some minor issues in the test implementation; specifically:

- The pixel buffer was too small
- Colors were being compared incorrectly due to color space differences; fix this by using the Util::compareColors
function instead.

* Tools/TestWebKitAPI/TestPDFDocument.swift:
(TestPDFPage.color(at:)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/DrawingToPDF.mm:
(TestWebKitAPI::TEST(DrawingToPDF, GradientIntoPDF)):
(TestWebKitAPI::TEST(DrawingToPDF, BackgroundClipText)):
(TestWebKitAPI::TEST(DrawingToPDF, SiteIsolationFormControl)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PDFSnapshot.mm:
(TestWebKitAPI::TEST(PDFSnapshot, FullContent)):
(TestWebKitAPI::TEST(PDFSnapshot, Subregions)):
(TestWebKitAPI::TEST(PDFSnapshot, Over200Inches)):
(TestWebKitAPI::TEST(PDFSnapshot, Links)):
(TestWebKitAPI::TEST(PDFSnapshot, AllowTransparentBackground)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewPrintFormatter.mm:
* Tools/TestWebKitAPI/Tests/ios/ApplicationStateTracking.mm:
* Tools/TestWebKitAPI/ios/TestPDFHostViewController.h:
* Tools/TestWebKitAPI/ios/TestPDFHostViewController.mm:

Canonical link: <a href="https://commits.webkit.org/296229@main">https://commits.webkit.org/296229@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b1ca82038ef2a429c04357628aa624e16c27da9b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107886 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27552 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17970 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113098 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58409 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109850 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28247 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36100 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81884 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110834 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22377 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97197 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62313 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21792 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15330 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57851 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91728 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15396 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116224 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34958 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25716 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90916 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35334 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93475 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90709 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23109 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35613 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13364 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30709 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34857 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34599 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37960 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36261 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/PDFSnapshot.mm:
```diff
@@ -173,7 +173,7 @@
 
         EXPECT_TRUE(CGRectEqualToRect([page bounds], CGRectMake(0, 0, 800, 14400)));
-        EXPECT_TRUE([[page colorAtPoint:NSMakePoint(400, 300)] isEqual:[CocoaColor whiteColor]]);
+        EXPECT_TRUE([[page colorAtPoint:NSMakePoint(400, 300)] isEqual:[CocoaColor colorWithRed:1.0 green:1.0 blue:1.0 alpha:1.0]]);
```
`compareColors` helper would make this less error prone and let you still use `whiteColor`

aprotyas replied:
And if you added Gtest assertion support, you could write:

```
EXPECT_EQ([page colorAtPoint:NSMakePoint(400, 300)], [CocoaColor whiteColor]);
```

... which would end up calling `compareColors` for you under the hood!

rr-codes committed:
Fix some PDF tests after 296104@main
https://bugs.webkit.org/show_bug.cgi?id=294482
rdar://153344673

Reviewed by Abrar Rahman Protyasha.

296104@main accidentally broke some PDF tests, but it also conveniently also accidentally
disabled those same tests because it didn't replace the PDFKIT_FOR_TESTING flag everywhere.

Fix that, and then fix some minor issues in the test implementation; specifically:

- The pixel buffer was too small
- Colors were being compared incorrectly due to color space differences; fix this by using the Util::compareColors
function instead.

* Tools/TestWebKitAPI/TestPDFDocument.swift:
(TestPDFPage.color(at:)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/DrawingToPDF.mm:
(TestWebKitAPI::TEST(DrawingToPDF, GradientIntoPDF)):
(TestWebKitAPI::TEST(DrawingToPDF, BackgroundClipText)):
(TestWebKitAPI::TEST(DrawingToPDF, SiteIsolationFormControl)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PDFSnapshot.mm:
(TestWebKitAPI::TEST(PDFSnapshot, FullContent)):
(TestWebKitAPI::TEST(PDFSnapshot, Subregions)):
(TestWebKitAPI::TEST(PDFSnapshot, Over200Inches)):
(TestWebKitAPI::TEST(PDFSnapshot, Links)):
(TestWebKitAPI::TEST(PDFSnapshot, AllowTransparentBackground)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewPrintFormatter.mm:
* Tools/TestWebKitAPI/Tests/ios/ApplicationStateTracking.mm:
* Tools/TestWebKitAPI/ios/TestPDFHostViewController.h:
* Tools/TestWebKitAPI/ios/TestPDFHostViewController.mm:

Canonical link: https://commits.webkit.org/296229@main
